fairlearn/reductions/_moments/regression_moment.py

```diff
@@ -0,0 +1,134 @@
+from .utility_parity import UtilityParity
+from .moment import Moment,_LABEL,_GROUP_ID
+from .weighted_error_rate import WeightedErrorRate
+from .bounded_group_loss import SquareLoss
+import pandas as pd
+import numpy as np
+from itertools import repeat
+
+class CDF_DemographicParity(Moment):
+    short_name = "CDF_DemographicParity"
+
+    def __init__(self,loss,y_range,difference_bound=None,grids=[],grid_num=41):
+        super().__init__()
+        self.utility_parity = UtilityParity(difference_bound=difference_bound)
+        self.loss = loss
+        self.objective = WeightedErrorRate(self.loss) #loss: SquareLoss, AbsoluteLoss, implemented inside the bounded_group_loss.py
+        self.grids = grids
+        self.grid_num = grid_num
+        self.y_range = y_range
+
+        #check the validity of y_range
+        assert (self.y_range is not None), "You should specify the range of y"
+        assert self.y_range[0] <= self.y_range[1], "The range of the label y is not valid!"
+        print(self.grids)
+        #determine the grids
+        if self.grids.size == 0:
+            self.grids = np.linspace(self.y_range[0],self.y_range[1],self.grid_num)
```

Comment:
you should try to run `flake8` in the repo root directory. I imagine it'll have a lot of things to complain about in terms of whitespace between args. The gates for the PR won't turn green otherwise.